src/core/timer.c

```diff
@@ -594,12 +596,23 @@ static void timer_enter_running(Timer *t) {
                 return;
         }
 
-        r = manager_add_job(UNIT(t)->manager, JOB_START, trigger, JOB_REPLACE, NULL, &error, NULL);
+        info = activation_event_info_new(UNIT(t));
+        if (!info) {
+                r = -ENOMEM;
+                goto fail;
+        }
+
+        r = manager_add_job(UNIT(t)->manager, JOB_START, trigger, JOB_REPLACE, NULL, &error, &job);
         if (r < 0)
                 goto fail;
 
         dual_timestamp_get(&t->last_trigger);
 
+        if (job) {
+                ACTIVATION_EVENT_INFO_TIMER(info)->last_trigger = t->last_trigger;
+                job->activation_event_info = activation_event_info_ref(info);
```

Comment:
here too. if the job object wasn't allocated new here, but result of job coalescing, there might already be the activation_event_info field be initialized.

maybe add a helper: job_set_activation_event_info(), that does the unreffing+reffing, and is then also the place where the policy of "oldest activation info wins" is singularly implemented (with comment)